### PR TITLE
Better Assets

### PR DESCRIPTION
**Pull request by holygits**
**Change Summary:**

- Display Asset symbol in AssetId type fields
Screen Shot 2021-03-29 at 16 04 33

- Pass through decimal place information in balance components allowing them to display dynamically

- use asset registry as source of truth for chain assets. Queries asset meta info once at API connect to register all assets.
 
- Balances with meaningless 0s e.g. `12345.0000` now display as `12,345` (required for 18dp assets where there can be many meaningless 0s)

### CODE REVIEW

File: packages/react-api/src/Api.tsx [hunk 1]
```diff
@@ -66,11 +70,13 @@ export function supportOldKeyringInLocalStorage() {
...
 
 async function loadOnReady (api: ApiPromise): Promise<State> {
-  const [properties, _systemChain, _systemName, _systemVersion, injectedAccounts] = await Promise.all([
+  const [properties, _systemChain, _systemName, _systemVersion, registeredAssets, injectedAccounts] = await Promise.all([
     api.rpc.system.properties(),
     api.rpc.system.chain(),
     api.rpc.system.name(),
     api.rpc.system.version(),
+    // @ts-ignore
+    api.rpc.genericAsset.registeredAssets() as Array<[AssetId, AssetInfo]>,
```
Comment (holygits) [inline packages/react-api/src/Api.tsx:79]: cennznet/api doesn't seem to expose RPC type defs

File: packages/react-components/src/AddressMini.tsx [hunk 2]
```diff
@@ -121,6 +121,10 @@ export default styled(AddressMini)`
     }
   }
 
+  .container.highlight::before {
```
Comment (holygits) [inline packages/react-components/src/AddressMini.tsx:124]: drive by fix to remove shadows from identicons

File: packages/react-params/src/index.tsx [hunk 3]
```diff
@@ -80,6 +82,14 @@ class Params extends React.PureComponent<Props, State> {
       return null;
     }
 
+    if (values && params) {
+      params.map(({ type }: ParamDef, index: number) => {
+        if (type.type?.includes('AssetId')) {
+          this.state.assetIdContext = values[index].value.toString();
```
Comment (holygits) [inline packages/react-params/src/index.tsx:88]: when a dynamic extrinsic or query component renders we track any asset ID fields and pass that to balance components later on

File: packages/app-accounts/src/AddressInfoMvp.tsx [hunk 4]
```diff
@@ -48,28 +47,9 @@ interface Props extends BareProps {
...
-      assetsRegistry.add(
-        spendingAssetId.toNumber().toString(),
-        SPENDING_ASSET_NAME
-      );
-    }
-
-  const cennzBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance as any, [assetsRegistry.getStakingAssetId(), address]);
-  const cpayBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance as any, [assetsRegistry.getSpendingAssetId(), address]);
+  const cennzBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance, [new AssetRegistry().getStakingAssetId(), address]);
+  const cpayBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance, [new AssetRegistry().getSpendingAssetId(), address]);
```
Comment (KarishmaBothara) [inline packages/app-accounts/src/AddressInfoMvp.tsx:52]: is it possible to create just one instance of AssetRegistry and then call
 const assetReg = new AssetRegistry();
const cennzBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance, [assetReg.getStakingAssetId(), address]);
   const cpayBalance = useCall<'Balance'>(api.query.genericAsset.freeBalance, [assetReg.getSpendingAssetId(), address]);

File: packages/react-params/src/index.tsx (hunk 3 above)
Comment (KarishmaBothara) [reply packages/react-params/src/index.tsx:88]: do we need to do 
`
this.setState({ assetIdContext: values[index].value.toString() });
`
to set the state

File: packages/app-accounts/src/AddressInfoMvp.tsx (hunk 4 above)
Comment (holygits) [reply packages/app-accounts/src/AddressInfoMvp.tsx:52]: AssetRegistry won't create a new instance as it's a singleton, I can change it too look like `AssetRegistry.instance()` instead

**Review by KarishmaBothara — APPROVED** [review]
lgtm

**Pull request merged**